### PR TITLE
Add GS main branch build CI testing

### PR DESCRIPTION
**Pull request by chrissimpkins**
Adds a GH Action workflow to build the GS project on pull request commits with a debug build of the fontc compiler.

### CODE REVIEW

**Review by rsheeter — APPROVED** [review]
Fantastic, please merge once it passes :)

File: .github/workflows/gsbuild.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          token: ${{ secrets.GS_READ_FONTC }}
+      - name: Install the latest stable Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - name: Install fontc and compile fonts
+        uses: googlefonts/fontc-action@v1
+        with:
+          source-path: source/GoogleSans/GoogleSans.designspace
+          debug: true
```
Comment (rsheeter) [inline .github/workflows/gsbuild.yml:24]: You could just assume we are in the root of our repo and run cargo commands akin to what https://github.com/googlefonts/fontc/blob/main/.github/workflows/rust.yml does?

File: .github/workflows/gsbuild.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  build-italic:
+    runs-on: ubuntu-latest
+    name: Build Italic variable font
```
Comment (rsheeter) [inline .github/workflows/gsbuild.yml:28]: If compiling the compiler is by far the slowest step  it's possible that building Romand and Italic in one job would make sense?

File: .github/workflows/gsbuild.yml (hunk 1 above)
Comment (chrissimpkins) [reply .github/workflows/gsbuild.yml:24]: yeah, I guess that was kind of silly, eh?

Comment (chrissimpkins) [reply .github/workflows/gsbuild.yml]: 7db4cb1f058d0a01e687312099dd784390dd904f

File: .github/workflows/gsbuild.yml (hunk 2 above)
Comment (chrissimpkins) [reply .github/workflows/gsbuild.yml]: 7db4cb1f058d0a01e687312099dd784390dd904f

Comment (chrissimpkins) [reply .github/workflows/gsbuild.yml]: With the current approach the fontc compile takes about 50s and each variable font compile takes 1.5 mins.  OK to keep this sequential approach or should we run the builds in parallel again?

**Pull request merged**